### PR TITLE
Eco 1214/new balance ui

### PR DESCRIPTION
**Pull request by OrenZak**
#### Main purpose:
Create new balance view component
#### Technical description:
- Updated BalanceView
- Updated Balance Presenter
- Started the Main page -> just for testing not as part of the main page task that will come after.
#### Tests:
- Updated BalancePreseterTest

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ jobs:
   build-pull-request:
     docker:
       # specify the version you desire here
-      - image: circleci/android:api-27-alpha
+      - image: circleci/android:api-26-alpha
```
Comment (drEfrati) [inline .circleci/config.yml:6]: Are you sure you want to work on something that is alpha?
Also why did you downgrade?

File: sdk/src/main/java/com/kin/ecosystem/balance/presenter/IBalancePresenter.java [hunk 2]
```diff
@@ -2,22 +2,9 @@
...
-
-		void onClick();
-	}
 
-	void onStart();
-
-	void onStop();
-
-	void balanceClicked();
+public interface IBalancePresenter extends IBasePresenter<IBalanceView> {
```
Comment (drEfrati) [inline sdk/src/main/java/com/kin/ecosystem/balance/presenter/IBalancePresenter.java:7]: 1. Why a marker(empty) interface?
2. where did all those method went?
For example, why did you move addObservers() to onAttach() and removeObservers() to onDetach?
What was the problem in the old way?
3. why you didn't made this interface also in Kotlin?

File: sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java [hunk 3]
```diff
@@ -43,9 +44,9 @@ public static MarketplaceFragment newInstance(INavigator navigator) {
 	private INavigator navigator;
 
 
-	private TextView spendSubTitle;
+//	private TextView spendSubTitle;
```
Comment (drEfrati) [inline sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java:47]: Please remove all comments, if you will want to revert it then you can always get it from the history

File: sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java [hunk 4]
```diff
@@ -123,7 +124,7 @@ public void onItemClick(BaseRecyclerAdapter adapter, View view, int position) {
 
 	@Override
 	public void setSpendList(List<Offer> spendList) {
-		spendRecyclerAdapter.setNewData(spendList);
+//		spendRecyclerAdapter.setNewData(spendList);
```
Comment (drEfrati) [inline sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java:127]: Same comment as above, if no need for this method them delete it,

File: sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java [hunk 5]
```diff
@@ -184,33 +184,33 @@ public void notifyEarnItemInserted(int index) {
 
 	@Override
 	public void notifySpendItemRemoved(int index) {
-		spendRecyclerAdapter.notifyItemRemoved(index);
+//		spendRecyclerAdapter.notifyItemRemoved(index);
```
Comment (drEfrati) [inline sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java:187]: All those method that now aren't doing anything, is it ok or you forgot to remove the comments?
If it is ok then how do you get updated and/or get notified by all the events?

File: sdk/src/main/java/com/kin/ecosystem/onboarding/view/OnboardingFragment.kt [hunk 6]
```diff
@@ -76,7 +76,7 @@ class OnboardingFragment: Fragment(), IOnboardingView {
         with(loadingAVD) {
             visibility = View.VISIBLE
             animate().alpha(AnimConsts.Value.ALPHA_1).setDuration(AnimConsts.Duration.FADE_ANIM).start()
-            (drawable as AnimatedVectorDrawable).start()
+            (drawable as AnimatedVectorDrawableCompat).start()
```
Comment (drEfrati) [inline sdk/src/main/java/com/kin/ecosystem/onboarding/view/OnboardingFragment.kt:79]: Just for my knowledge, why did you went to Compat instead of the previous one?

File: sdk/src/main/java/com/kin/ecosystem/onboarding/view/OnboardingFragment.kt [hunk 7]
```diff
@@ -93,7 +93,7 @@ class OnboardingFragment: Fragment(), IOnboardingView {
                         .setDuration(AnimConsts.Duration.FADE_ANIM)
                         .withStartAction {
                             visibility = View.INVISIBLE
-                            (drawable as AnimatedVectorDrawable).stop()
+                            (drawable as AnimatedVectorDrawableCompat).stop()
```
Comment (drEfrati) [inline sdk/src/main/java/com/kin/ecosystem/onboarding/view/OnboardingFragment.kt:96]: Will come and ask you later what is this drawable and what is the "with" keyword

**Review by drEfrati — APPROVED** [review]

**Comment by drEfrati** [comment]
Overall it seems good but I am not Kotlin expert(yet) so just make sure you test it manually and with unit/integration tests.

File: .circleci/config.yml (hunk 1 above)
Comment (OrenZak) [reply .circleci/config.yml:6]: we had some issues because our build SDK is with API 26

File: sdk/src/main/java/com/kin/ecosystem/balance/presenter/IBalancePresenter.java (hunk 2 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/balance/presenter/IBalancePresenter.java:7]: 1. to be aligned with all views
2. Before the change the presenter was initialized on the EcoActivity, now it's on the BalanceView itself.
3. not sure will do :)

File: sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java (hunk 4 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java:127]: it was only for testing, next task is handling all of this so, I didn't bother.

File: sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java (hunk 5 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java:187]: will be one list

File: sdk/src/main/java/com/kin/ecosystem/onboarding/view/OnboardingFragment.kt (hunk 6 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/onboarding/view/OnboardingFragment.kt:79]: to support below API 21

File: sdk/src/main/java/com/kin/ecosystem/onboarding/view/OnboardingFragment.kt (hunk 7 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/onboarding/view/OnboardingFragment.kt:96]: 👍
